src/zarr/v3/group.py

```diff
@@ -377,7 +420,7 @@ def __setitem__(self, key, value):
         """__setitem__ is not supported in v3"""
         raise NotImplementedError
 
-    async def update_attributes_async(self, new_attributes: Dict[str, Any]) -> Group:
+    async def update_attributes_async(self, new_attributes: dict[str, Any]) -> Group:
```

Comment:
I'm a bit surprised that these `_async` methods are still on the Group class. Did I miss these in #1590 / #1634? 

In any event, not your problem but something I should revisit.